src/game/mario_misc.c

```diff
@@ -183,38 +183,50 @@ void bhv_toad_message_loop(void) {
 
 void bhv_toad_message_init(void) {
     s32 saveFlags = save_file_get_flags();
+#ifndef UNLOCK_ALL
```

Comment:
if you were to make starCount be a massive number (or just 120) then you could remove those other ifdefs to keep the code clean.. plus gcc would likely optimize out the checks against the constants.